hasher-matcher-actioner/hmalib/common/content_models.py

```diff
@@ -226,6 +214,37 @@ def to_dynamodb_item(self) -> dict:
             },
         )
 
+    def write_to_table_if_not_found(self, table: Table) -> bool:
+        try:
+            table.put_item(
+                Item={
+                    "PK": self.get_dynamodb_content_key(self.content_id),
+                    "SK": self.get_dynamodb_content_type_key(),
+                    "ContentType": self.content_type.get_name(),
+                    "ContentRef": self.content_ref,
+                    "ContentRefType": self.content_ref_type.value,
+                    "AdditionalFields": self.additional_fields
+                    if self.additional_fields
+                    else {self.ADDITIONAL_FIELDS_PLACE_HOLDER},
+                    "SubmissionTimes": [s.isoformat() for s in self.submission_times],
+                    "CreatedAt": self.created_at.isoformat(),
+                    "UpdatedAt": self.updated_at.isoformat(),
+                },
+                ConditionExpression="attribute_not_exists(PK) AND attribute_not_exists(SK)",
+            )
+        except ClientError as client_error:
+            # boto3 exception handling https://imgflip.com/i/5f5zfj
```

Comment:
Look what we've been reduced to! I actually followed the link!

Reply:
you did this to yourself: https://github.com/facebook/ThreatExchange/blame/main/hasher-matcher-actioner/hmalib/models.py#L39